data_generation/generate.py

```diff
@@ -117,53 +120,58 @@ def create_networks(arguments):
-        if arguments.n_1:
-            subgraph_busses = list(full_net.bus.index)
-            downed_bus = np.random.randint(0, len(subgraph_busses))
-            del subgraph_busses[downed_bus]
-    
+        if(arguments.subgraphing_method == 'num_change'):
+            subgraph_net = subgraphing_method(copy.deepcopy(full_net), None, None)
+            solve_and_save(subgraph_net,arguments,len(subgraph_net.bus),full_net,None)
+            i += 1
         else:
-            subgraph_busses = subgraphing_method(full_net, initial_bus, subgraph_length)
+            subgraph_length = np.random.randint(arguments.min_size, min(arguments.max_size, len(full_net.bus)))
+            initial_bus = starting_points[np.random.randint(0, len(starting_points))]
+            
+            if arguments.n_1:
+                subgraph_busses = list(full_net.bus.index)
+                downed_bus = np.random.randint(0, len(subgraph_busses))
+                del subgraph_busses[downed_bus]
         
-        subgraph_net = tb.select_subnet(full_net, subgraph_busses)
-
-        try:
-            # check if the subgraph contains a slack bus, if not add one by setting the slack bus to a random bus
-            if full_net.ext_grid.bus.item() not in subgraph_busses:
-                slack_bus = subgraph_busses[np.random.randint(0, len(subgraph_busses))]
-                # https://pandapower.readthedocs.io/en/v2.1.0/elements/ext_grid.html#pandapower.create_ext_grid
-                pp.create_ext_grid(subgraph_net, slack_bus)
-                
-            pp.runpp(subgraph_net)
-            # ppl.simple_plot(subgraph_net, plot_loads=True, plot_gens=True, trafo_color="r", switch_color="g") 
-
-        except:
-            print(f"Network not solvable trying a new one")
-            continue
-
-        uid = ''.join([random.choice(string.ascii_letters
-                + string.digits) for _ in range(8)])
-        
-        Path(f"{arguments.save_dir}/x").mkdir(parents=True, exist_ok=True)
-        Path(f"{arguments.save_dir}/y").mkdir(parents=True, exist_ok=True)
-        
-        pp.to_json(subgraph_net, f"{arguments.save_dir}/x/{arguments.network}_{subgraph_length}_{arguments.subgraphing_method}_{uid}.json")
-        subgraph_net.res_gen.to_csv(f"{arguments.save_dir}/y/{arguments.network}_{subgraph_length}_{arguments.subgraphing_method}_{uid}_gen.csv")
-        subgraph_net.res_line.to_csv(f"{arguments.save_dir}/y/{arguments.network}_{subgraph_length}_{arguments.subgraphing_method}_{uid}_line.csv")
-        subgraph_net.res_bus.to_csv(f"{arguments.save_dir}/y/{arguments.network}_{subgraph_length}_{arguments.subgraphing_method}_{uid}_bus.csv")
+            else:
+                subgraph_busses = subgraphing_method(full_net, initial_bus, subgraph_length)
+            
+            subgraph_net = tb.select_subnet(full_net, subgraph_busses)
+            solve_and_save(subgraph_net,arguments,subgraph_length,full_net,subgraph_busses)
 
-        i += 1
+            i += 1
```

Comment:
why is this shifted a tab

Reply:
It was a call inside the If statement, and it's else. But was used in both so now has been moved out.
Good spotting.